### ISSUE TITLE
[Feature Request] XDG Path Support

### ISSUE THREAD

**Issue by jeffgazso** [dvehrs/podget#63]
Please consider adding XDG path support to podget. Use of XDG paths by default is generally considered best practice these days and it cuts down on dot file/directory clutter in one's own home directory. At a glance, it appears this would be as simple as checking to see if `${XDG_CONFIG_HOME}` is set and, if so, swapping `DIR_CONFIG="${HOME}/.podget"` for `DIR_CONFIG="${XDG_CONFIG_HOME}/podget"`. Thoughts?

**Comment by dvehrs** [comment]
Interesting idea but I will have to think about it.

First issue, XDG_CONFIG_HOME is not set on any of my systems (Debian Sid with Sway WM or Debian Bullseye with console sessions).  Now this is a minor issue but highlights that it's use may not be common enough to really be considered a best practice.  Having said that it's probably not a bad idea to reduce the number of dot-files in a users home directory.  The mere quantity of those can get overwhelming.  Now my systems do not have XDG_CONFIG_HOME set but do have a significant number of applications that are using ${HOME}/.config to store configuration files.  Maybe this needs to be a multiple step process.  First check if XDG_CONFIG_HOME is set, if so use it.  If not, check if .config exists, then use ${HOME}/.config/podget.  For new users, do we stop here and if ${HOME}/.config does not exist then we create it and put our configuration files in it?  If we are creating the directories then we can add what we like.

Second issue, if we do this we will need to find a way that is compatible with all of the platforms were Podget is expected to run.  I don't know how common the use of the XDG directories is for the BSDs or Mac OS.  However this issue might be addressed by the testing our way down the chain of possible locations for configuration files.  So it's fix might be the same as what is needed for the first.   

Third issue, migrating existing users forward.  Do we make this a change that only affects new installs?  If we find a user with "${HOME}/.podget"  but XDG_CONFIG_HOME is set or ${HOME}/.config exists, do we allow it to continue but alert the user that the configuration should be moved?  This third issue may be the most complicated, because we can't simply start with a configured XDG_CONFIG_HOME and assume that if a podget folder does not exist in there, that we can simply create new ones and move on.  To support existing users, we will need to test all the way down the chain for existing configuration files and then use what they have.  

Final issue, what changes do we need to make to the documentation and manual pages so that users can find everything?

Now this leads to step two of this conversion.  Are there other directories that should be changed?

Should the default directory for downloaded files be changed from ${HOME}/POD to XDG_DATA_HOME/POD ?  This could add a few complexities because we would need to do similar tests for what is configured and exists before we created the initial versions of the podgetrc file.  And is this a check we only do when the podgetrc is created or should it be done on every run to alert the user to new possibilities?  

Should the session file be moved from ${TMPDIR}/podget (or /tmp/podget) to XDG_RUNTIME_DIR/poget ?  Now my systems do have this one set so from that small sample size that appears to be closer to a best practice but then we get into the question of what is best for the BSDs or Mac OS.  Is this just a matter of adding one more level to the testing of where will we store a few temporary files?

**Comment by bc-spreedly** [comment]
@dvehrs , give the [spec](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) a read. Nobody's XDG_* is set by default - applications are expected to honor the environment variables if specified, and falling back to the standard locations. if not.

A pretty standard migration path is to check for $HOME/.podget, and if that doesn't exist, use the XDG directory specification.

**Comment by bc-spreedly** [comment]
Another note is that using the XDG dir spec doesn't preclude setting a directory in the user's home directory (thinking of the $HOME/POD directory). I'd say that data *not related to the end-user-consumed content* belongs in the $XDG_DATA_HOME directory.

**Comment by dvehrs** [comment]
@bc-spreedly 

> @dvehrs, those are good questions. I think it would help if you give the [spec](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) a read. Nobody's XDG_* is set by default - applications are expected to honor the environment variables if specified, and falling back to the standard locations. if not.
> 
> A pretty standard migration path is to check for $HOME/.podget, and if that doesn't exist, use the XDG directory specification.

I've read the spec and if I was writing a program from scratch, I would definitely keep it in mind.  However Podget is not new and all changes need to be made so they don't break things for existing users.

The "standard migration path" does not address any of the existing user issues.  In this case, it appears that supporting existing users will add complexity to the fix.  So far I have not come up with an elegant way to handle it all but I am open to consider any patches that do.

**Comment by dvehrs** [comment]
OK, I think I might have a working solution.    I've uploaded it to the DEV branch.

For existing users, it looks for a configuration directory in:

1.  $HOME/.podget
2. XDG_CONFIG_HOME/podget
3. $HOME/.config/podget

It goes through them in order, first found wins.

For new users, it attempts to create a new configuration directory in:

1. $XDG_CONFIG_HOME/podget
2. $HOME/.config/podget

In order and only moves on to number 2 if XDG is undefined.

This has only had a little bit of testing so there is a bunch to go before it can be merged with MASTER.

There might also be a couple small unrelated fixes that can also be applied now.

Try it out, let me know if you have any issues.

**Comment by ainola** [comment]
I installed the new version from the `dev` branch. I can confirm that it works as expected!

* When I have a $HOME/.podget directory it reads the config from there
* When I _don't_ have a $HOME/.podget dir but do have .config/podget it sources the latter
* If I have neither directory, it creates .config/podget.
* When I set XDG_CONFIG_HOME=asdf then it complains gracefully that there's no directory. If it does exist it creates asdf/podget as expected.

I can't see any issues with this. Great work!
